workitem/workitem_repository.go

```diff
@@ -95,6 +97,43 @@ func (r *GormWorkItemRepository) Load(ctx context.Context, ID string) (*app.Work
 	return convertWorkItemModelToApp(wiType, res)
 }
 
+// LoadTopWorkitem returns top most work item of the list. Top most workitem has the Highest order.
+// returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemRepository) LoadTopWorkitem(ctx context.Context) (*app.WorkItem, error) {
+	res := WorkItem{}
+	r.db = r.db.Debug()
+	tx := r.db.Order("execution_order desc").Last(&res)
+	if tx.RecordNotFound() {
+		return nil, nil
```

Comment:
Here you return a `nil` error but in `LoadBottomWorkitem` you return the `tx.Error`. I'd say you should return a `errors.NewNotFoundError` in both cases, no?